src/main/java/org/apache/commons/text/numbers/DoubleFormat.java

```diff
@@ -0,0 +1,736 @@
+         *  <li>{@link #decimalSeparator(char) decimal separator}</li>
+         *  <li>{@link #groupingSeparator(char) thousands grouping separator}</li>
+         *  <li>{@link #minusSign(char) minus sign}</li>
+         *  <li>{@link #exponentSeparator(String) exponent separator}</li>
+         *  <li>{@link #infinity(String) infinity}</li>
+         *  <li>{@link #nan(String) NaN}</li>
+         * </ul>
+         * The digit character string is constructed by starting at the configured
+         * {@link DecimalFormatSymbols#getZeroDigit() zero digit} and adding the next
+         * 9 consecutive characters.
+         * @param symbols format symbols
+         * @return this instance
+         * @throws NullPointerException if the argument is null
+         */
+        public Builder formatSymbols(final DecimalFormatSymbols symbols) {
+            Objects.requireNonNull(symbols, "Decimal format symbols cannot be null");
+
+            return digits(getDigitString(symbols))
+                    .decimalSeparator(symbols.getDecimalSeparator())
+                    .groupingSeparator(symbols.getGroupingSeparator())
+                    .minusSign(symbols.getMinusSign())
+                    .exponentSeparator(symbols.getExponentSeparator())
+                    .infinity(symbols.getInfinity())
+                    .nan(symbols.getNaN());
+        }
+
+        /** Get a string containing the localized digits 0-9 for the given symbols object. The
+         * string is constructed by starting at the {@link DecimalFormatSymbols#getZeroDigit() zero digit}
+         * and adding the next 9 consecutive characters.
+         * @param symbols symbols object
+         * @return string containing the localized digits 0-9
+         */
+        private String getDigitString(final DecimalFormatSymbols symbols) {
+            final int zeroDelta = symbols.getZeroDigit() - DEFAULT_DECIMAL_DIGITS.charAt(0);
+
+            final char[] digitChars = new char[DEFAULT_DECIMAL_DIGITS.length()];
+            for (int i = 0; i < DEFAULT_DECIMAL_DIGITS.length(); ++i) {
+                digitChars[i] = (char) (DEFAULT_DECIMAL_DIGITS.charAt(i) + zeroDelta);
+            }
+
+            return String.valueOf(digitChars);
+        }
+
+        /** Construct a new double format function.
+         * @return format function
+         */
+        public DoubleFunction<String> build() {
+            return factory.apply(this);
+        }
+    }
+
+    /** Base class for standard double formatting classes.
+     */
+    private abstract static class AbstractDoubleFormat
+        implements DoubleFunction<String>, ParsedDecimal.FormatOptions {
+
+        /** Maximum precision; 0 indicates no limit. */
+        private final int maxPrecision;
+
+        /** Minimum decimal exponent. */
+        private final int minDecimalExponent;
+
+        /** String representing positive infinity. */
+        private final String postiveInfinity;
```

Comment:
By chance I was debugging an example experimenting with a builder and a custom character (`~`) for minus sign, when I spotted this typo (`s/postiveInfinity/positiveInfinity`)

Reply:
Good catch.